mesonbuild/build.py

```diff
@@ -2388,6 +2396,9 @@ def get_debug_filename(self) -> T.Optional[str]:
         """
         return self.debug_filename
 
+    def get_toplevel_link_deps(self):
+        return [self] + self.link_targets
```

Comment:
An extra optimisation would be to return a itertool.chain() iterator instead of constructing a new list here. Also it should contain self.link_whole_targets.

Reply:
Same here - using link_whole_targets would change behavior from what master is doing.